### PR TITLE
hdrgen: remove trailing whitespace on attribute declarations

### PR DESCRIPTION
**Pull request by ljmf00**
Fixes Issue 22353.

Signed-off-by: Luís Ferreira <contact@lsferreira.net>

### CODE REVIEW

File: src/dmd/hdrgen.d [hunk 1]
```diff
@@ -918,10 +924,12 @@ public:
         if (d.decl.dim == 0 || (hgs.hdrgen && d.decl.dim == 1 && (*d.decl)[0].isUnitTestDeclaration()))
         {
             // hack for bugzilla 8081
+            if(hasSTC) buf.writeByte(' ');
```
Comment (Geod24) [inline src/dmd/hdrgen.d:927]:
```suggestion
            if (hasSTC) buf.writeByte(' ');
```

File: src/dmd/hdrgen.d [hunk 2]
```diff
@@ -918,10 +924,12 @@ public:
...
             buf.writestring("{}");
         }
         else if (d.decl.dim == 1)
         {
+            if(hasSTC) buf.writeByte(' ');
```
Comment (Geod24) [inline src/dmd/hdrgen.d:932]:
```suggestion
            if (hasSTC) buf.writeByte(' ');
```

Comment (ljmf00) [reply src/dmd/hdrgen.d]: Done

File: src/dmd/hdrgen.d (hunk 1 above)
Comment (ljmf00) [reply src/dmd/hdrgen.d]: Done

**Review by thewilsonator — APPROVED** [review]

**Review by WalterBright — APPROVED** [review]

**Comment by thewilsonator** [comment]
please rebase to restart CI

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**